CRM/Eck/BAO/EckEntityType.php

```diff
@@ -29,25 +29,31 @@ class CRM_Eck_BAO_EckEntityType extends CRM_Eck_DAO_EckEntityType implements Hoo
    */
   public static function getEntityTypes(): array {
     $entityTypes = Civi::cache('metadata')->get('EckEntityTypes');
+    // The table might not yet exist (e.g. when flushing caches/fetching permissions during installation).
     if (!is_array($entityTypes)) {
-      $entityTypesQuery = CRM_Core_DAO::executeQuery('SELECT * FROM `civicrm_eck_entity_type`');
-      if (!is_a($entityTypesQuery, CRM_Core_DAO::class)) {
-        throw new CRM_Core_Exception('Error retrieving ECK entity types: ');
+      if (CRM_Core_DAO::checkTableExists('civicrm_eck_entity_type')) {
+        $entityTypesQuery = CRM_Core_DAO::executeQuery('SELECT * FROM `civicrm_eck_entity_type`');
+        if (!is_a($entityTypesQuery, CRM_Core_DAO::class)) {
+          throw new CRM_Core_Exception('Error retrieving ECK entity types: ');
+        }
+        $entityTypes = $entityTypesQuery->fetchAll();
+        foreach ($entityTypes as &$entityType) {
+          $entityType['entity_name'] = 'Eck_' . $entityType['name'];
+          $entityType['table_name'] = _eck_get_table_name($entityType['name']);
+        }
+        $entityTypes = array_combine(
+          array_map(
+            function($entityType) {
```

Comment:
I think an arrow function would look nicer in this case.

Reply:
Refactored this using `CRM_Core_DAO->fetch()` instead …